datafusion/src/physical_plan/hash_join.rs

```diff
@@ -65,7 +65,7 @@ use log::debug;
 // Maps a `u64` hash value based on the left ["on" values] to a list of indices with this key's value.
 // E.g. 1 -> [3, 6, 8] indicates that the column values map to rows 3, 6 and 8 for hash value 1
 // As the key is a hash value, we need to check possible hash collisions in the probe stage
-type JoinHashMap = HashMap<u64, SmallVec<[u64; 1]>, IdHashBuilder>;
+type JoinHashMap = HashMap<(), SmallVec<[u64; 1]>, IdHashBuilder>;
```

Comment:
I am a little confused about using `()` as the key type for the `HashMap` -- maybe updating the comments would help

Reply:
I will add some comments about it for sure!

Eventually we can get rid of the hashmap entirely, as we check for collisions in a later stage.

Reply:
Added some extra docs now :)